sdk/loadtestservice/load-testing-rest/src/pollingHelper.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzureLoadTestingClient } from "./clientDefinitions";
+import { getFileValidationPoller } from "./getFileValidationPoller";
+import { getTestRunCompletionPoller } from "./getTestRunCompletionPoller";
+import {
+  FileUploadAndValidatePoller,
+  TestUploadFileSuccessResponse,
+  TestRunCompletionPoller,
+  TestRunCreateOrUpdateSuccessResponse,
+} from "./models";
+
+export async function getLongRunningPoller(
+  client: AzureLoadTestingClient,
+  initialResponse: TestUploadFileSuccessResponse
+): Promise<FileUploadAndValidatePoller | undefined>;
+export async function getLongRunningPoller(
+  client: AzureLoadTestingClient,
+  initialResponse: TestRunCreateOrUpdateSuccessResponse
+): Promise<TestRunCompletionPoller | undefined>;
+export async function getLongRunningPoller(
+  client: AzureLoadTestingClient,
+  initialResponse: TestRunCreateOrUpdateSuccessResponse | TestUploadFileSuccessResponse
+): Promise<TestRunCompletionPoller | FileUploadAndValidatePoller | undefined> {
+  if (isFileUpload(initialResponse)) {
+    return getFileValidationPoller(client, initialResponse);
+  } else if (isTestRunCreation(initialResponse)) {
+    return getTestRunCompletionPoller(client, initialResponse);
+  }
+  return undefined;
```

Comment:
I think we can throw in this case instead of returning an undefined because this case is very unlikely to happen because the types guarantee that the input must be of the correct type. So let's just throw and remove `undefined` from the return type.

Reply:
Updated.